SchedulearnBackend/SchedulearnBackend/Services/TeamService.cs

```diff
@@ -10,21 +13,60 @@ namespace SchedulearnBackend.Services
     public class TeamService
     {
         private readonly SchedulearnContext _schedulearnContext;
+        private readonly LimitService _limitService;
 
-        public TeamService(SchedulearnContext schedulearnContext)
+        public TeamService(LimitService limitService, SchedulearnContext schedulearnContext)
         {
+            _limitService = limitService;
             _schedulearnContext = schedulearnContext;
         }
 
-        public async Task<Team> AddNewTeamAsync(int managerId, int limitId) {
-            Team newTeam = new Team()
-            {
-                LimitId = limitId,
-                ManagerId = managerId
-            };
+        public async Task<List<Team>> AllTeamsAsync()
+        {
+            return await _schedulearnContext.Teams.ToListAsync();
+        }
+
+        public async Task<Team> GetTeamAsync(int id)
+        {
+            var team = await _schedulearnContext.Teams.FindAsync(id);
+            return team ?? throw new NotFoundException($"Team with id ({id}) does not exist");
+        }
+
+        public async Task<Team> AddNewTeamAsync(CreateNewTeam teamToCreate) {
+            var newTeam = teamToCreate.CreateTeam();
 
             await _schedulearnContext.Teams.AddAsync(newTeam);
+            await _schedulearnContext.SaveChangesAsync();
             return newTeam;
         }
+
+        public async Task<Team> ChangeLimitsForTeamAsync(int teamId, LimitsToApply limits)
+        {
+            var team = await GetTeamAsync(teamId);
+            var wantedLimit = await _limitService.GetLimitAsync(limits.LimitId);
+
+            team.LimitId = wantedLimit.Id;
+            _schedulearnContext.Update(team);
+            await _schedulearnContext.SaveChangesAsync();
+
+            return team;
+        }
+
+        public async Task<List<Team>> GetAccessibleTeams(int teamId)
+        {
+            var accessibleTeams = new List<Team>();
+            var team = await GetTeamAsync(teamId);
```

Comment:
We might need to think on how to optimize this in the future, since it's going to generate a lot of calls to DB.